instances/azp-build-asg/init.sh.tpl

```diff
@@ -15,9 +15,45 @@ hostnamectl set-hostname "$instance_id"
 
 azp_token=$(aws s3 cp s3://cncf-envoy-token/azp_token -)
 
-export AWS_DEFAULT_REGION=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | jq -r .region)
-aws ec2 replace-iam-instance-profile-association --iam-instance-profile Arn=${instance_profile_arn} \
-  --association-id $(aws ec2 describe-iam-instance-profile-associations --filter=Name=instance-id,Values=$instance_id | jq -r '.IamInstanceProfileAssociations[0].AssociationId')
+# Configured vars
+asg_name="${asg_name:-}"
+azp_pool_name="${azp_pool_name:-}"
+bazel_cache_bucket="${bazel_cache_bucket:-}"
+cache_prefix="${cache_prefix:-}"
+instance_profile_arn="${instance_profile_arn:-}"
+role_name="${role_name:-}"
+
+if [[ -n "$asg_name" ]]; then
+    echo "\`asg_name\` is not set, exiting" >&2
+    exit 1
+fi
+if [[ -n "$azp_pool_name" ]]; then
+    echo "\`azp_pool_name\` is not set, exiting" >&2
+    exit 1
+fi
+if [[ -n "$bazel_cache_bucket" ]]; then
+    echo "\`bazel_cache_bucket\` is not set, exiting" >&2
+    exit 1
+fi
+if [[ -n "$cache_prefix" ]]; then
+    echo "\`cache_prefix\` is not set, exiting" >&2
+    exit 1
+fi
+if [[ -n "$instance_profile_arn" ]]; then
+    echo "\`instance_profile_arn\` is not set, exiting" >&2
+    exit 1
+fi
+if [[ -n "$role_name" ]]; then
+    echo "\`role_name\` is not set, exiting" >&2
+    exit 1
+fi
+
+AWS_DEFAULT_REGION=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | jq -r .region)
```

Comment:
Curious - what is the reason to separate variable assignment from env var export into two statements.

E.g.
```
export AWS_DEFAULT_REGION="..."
```

Vs.

```
AWS_DEFAULT_REGION=""
export AWS_DEFAULT_REGION
```

Reply:
to prevent masking errors

i should say tho - im probs going to close this in favour of #33 which has a much cleaner solution